@types/index.d.ts

```diff
@@ -331,9 +331,37 @@ interface IOntoUMLError {
 interface IOntoUML2GUFOOptions {
   baseIRI: string;
   createInverses?: boolean;
-  format?: string;
   createObjectProperty?: boolean;
+  customElementMapping?: {
+    [key: string]: {
+      label?: {
+        [key: string]: string;
+      };
+      uri: string;
+    };
+  };
+  customPackageMapping?: {
+    [key: string]: {
+      prefix: string;
+      uri: string;
+    };
+  };
+  format?: string;
+  preAnalysis?: boolean;
   prefixPackages?: boolean;
   uriFormatBy?: 'name' | 'id';
   uriManager?: URIManager;
 }
+
+interface IPreAnalysisItem {
```

Comment:
I believe I can't have all the structure because the pre-analysis is a little different. I prefer keep the `meta` because we can have different stuff there (uri, baseIRI, element, etc.). I also think we should use lowercase for `severity` and `code` @claudenirmf

Reply:
Hey. Pre Analysis have some issues not related with IElement. For example a wrong baseIRI or prefix. Where we can add those cases? I can put it in source but we will need change the type.

Reply:
We discussed this issue offline and jointly arrived at a common message interface:

```typescript
{
  id: string
  code: string
  title: string
  description: string
  severity: string
  data: { alternatives, element, context, ... }
}
```
The fields `id`, `code`, `title`, `description`, and `severity` are used by all services.

`data` can be used by a service to add specific properties it requires